src/compute/flexbox.rs

```diff
@@ -339,45 +357,52 @@ fn compute_preliminary(
     // We have the container size.
     // If our caller does not care about performing layout we are done now.
     if run_mode == RunMode::ComputeSize {
-        let container_size = constants.container_size;
-        return container_size;
+        return SizeAndBaselines { size: constants.container_size, first_baselines: Point::NONE };
     }
 
     // 16. Align all flex lines per align-content.
     #[cfg(feature = "debug")]
     NODE_LOGGER.log("align_flex_lines_per_align_content");
-    align_flex_lines_per_align_content(tree, &mut flex_lines, node, &constants, total_line_cross_size);
+    align_flex_lines_per_align_content(tree, &mut flex_lines, &constants, total_line_cross_size);
 
     // Do a final layout pass and gather the resulting layouts
     #[cfg(feature = "debug")]
     NODE_LOGGER.log("final_layout_pass");
-    final_layout_pass(tree, node, &mut flex_lines, &constants);
+    final_layout_pass(tree, &mut flex_lines, &constants);
 
     // Before returning we perform absolute layout on all absolutely positioned children
     #[cfg(feature = "debug")]
     NODE_LOGGER.log("perform_absolute_layout_on_absolute_children");
-    perform_absolute_layout_on_absolute_children(tree, node, &constants);
+    perform_absolute_layout_on_absolute_children(tree, &constants);
 
     #[cfg(feature = "debug")]
     NODE_LOGGER.log("hidden_layout");
-    let len = tree.child_count(node);
+    let len = tree.child_count();
     for order in 0..len {
-        let child = tree.child(node, order);
-        if tree.style(child).display == Display::None {
-            *tree.layout_mut(node) = Layout::with_order(order as u32);
-            compute_node_layout(
-                tree,
-                child,
-                Size::NONE,
-                Size::NONE,
-                Size::MAX_CONTENT,
-                RunMode::PeformLayout,
-                SizingMode::InherentSize,
-            );
+        let child = tree.child(order);
+        if tree.child_style(child).display == Display::None {
+            *tree.child_layout_mut(child) = Layout::with_order(order as u32);
+            tree.measure_child_size(child, Size::NONE, Size::NONE, Size::MAX_CONTENT, SizingMode::InherentSize);
         }
     }
 
-    constants.container_size
+    // 8.5. Flex Container Baselines: calculate the flex container's first baseline
+    // See https://www.w3.org/TR/css-flexbox-1/#flex-baselines
+    let first_vertical_baseline = if flex_lines.is_empty() {
```

Comment:
I assume horizontal baselines are only relevant when supporting vertical text directions?

Reply:
Ah, well I'd phrase it as "vertical baselines are only relevant when supporting vertical text directions", but I suspect that makes the short answer "yes".